Not allowed to start service Intent: … : app is in background

Despite https://github.com/werman/RadioDroid/commit/ad607f8116991e8da3e488646530fca93f132e4e I'm still getting this error about once a day:

```
[04-25 23:56:00.542 3935:3935 E/AndroidRuntime]
FATAL EXCEPTION: main
Process: net.programmierecke.radiodroid2, PID: 3935
java.lang.RuntimeException: Unable to start activity ComponentInfo{net.programmierecke.radiodroid2/net.programmierecke.radiodroid2.ActivityMain}: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=net.programmierecke.radiodroid2/.service.PlayerService (has extras) }: app is in background uid UidRecord{5479c41 u0a411 TPSL idle procs:1 seq(0,0,0)}
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3270)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=net.programmierecke.radiodroid2/.service.PlayerService (has extras) }: app is in background uid UidRecord{5479c41 u0a411 TPSL idle procs:1 seq(0,0,0)}
	at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1616)
	at android.app.ContextImpl.startService(ContextImpl.java:1571)
	at android.content.ContextWrapper.startService(ContextWrapper.java:669)
	at net.programmierecke.radiodroid2.service.PlayerServiceUtil.bind(PlayerServiceUtil.java:47)
	at net.programmierecke.radiodroid2.ActivityMain.onCreate(ActivityMain.java:168)
	at android.app.Activity.performCreate(Activity.java:7824)
	at android.app.Activity.performCreate(Activity.java:7813)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1306)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
	... 11 more
```
I have not yet examined it in detail again, but I wanted to document it here.

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -45,7 +45,8 @@
             <!--</intent-filter>-->
         </activity>
         <service
-            android:name="net.programmierecke.radiodroid2.service.PlayerService">
+            android:name="net.programmierecke.radiodroid2.service.PlayerService"
+            android:permission="android.permission.BIND_JOB_SERVICE">
             <intent-filter>
                 <action android:name="android.intent.action.MEDIA_BUTTON" />
             </intent-filter>

--- a/app/src/main/java/net/programmierecke/radiodroid2/service/PlayerService.java
+++ b/app/src/main/java/net/programmierecke/radiodroid2/service/PlayerService.java
@@ -32,6 +32,8 @@ import android.os.PowerManager;
 import android.os.RemoteException;
 import android.support.v4.media.MediaMetadataCompat;
 
+import androidx.annotation.NonNull;
+import androidx.core.app.JobIntentService;
 import androidx.media.app.NotificationCompat.MediaStyle;
 import androidx.media.session.MediaButtonReceiver;
 
@@ -75,7 +77,7 @@ import net.programmierecke.radiodroid2.recording.RunningRecordingInfo;
 
 import static android.content.Intent.ACTION_MEDIA_BUTTON;
 
-public class PlayerService extends Service implements RadioPlayer.PlayerListener {
+public class PlayerService extends JobIntentService implements RadioPlayer.PlayerListener {
     protected static final int NOTIFY_ID = 1;
     private static final String NOTIFICATION_CHANNEL_ID = "default";
 
@@ -1283,5 +1285,10 @@ public class PlayerService extends Service implements RadioPlayer.PlayerListener
                 }
             });
         }
+    }
+
+    @Override
+    protected void onHandleWork(@NonNull Intent intent) {
+        Log.d(TAG, "onHandleWork called with intent: " + intent.toString());
     }
 }

